go-controller/pkg/ovn/multicast.go

```diff
@@ -236,16 +252,150 @@ func (oc *DefaultNetworkController) createDefaultAllowMulticastPolicy() error {
 	return nil
 }
 
+func (oc *DefaultNetworkController) disableMulticast() error {
+	// default mcast acls have MulticastGlobalOwnerType
+	predicateIDs := libovsdbops.NewDbObjectIDs(libovsdbops.ACLMulticastGlobal, oc.controllerName, nil)
+	mcastAclPred := libovsdbops.GetPredicate[*nbdb.ACL](predicateIDs, nil)
+	mcastACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, mcastAclPred)
+	if err != nil {
+		return fmt.Errorf("unable to find default multicast ACLs: %v", err)
+	}
+	err = libovsdbops.DeleteACLsFromPortGroups(oc.nbClient, []string{types.ClusterRtrPortGroupName, types.ClusterPortGroupName},
+		mcastACLs...)
+	if err != nil {
+		return fmt.Errorf("unable to delete default multicast acls: %v", err)
+	}
+	// run sync for empty namespaces list, this should delete namespaces objects
+	err = oc.syncNsMulticast(map[string]bool{})
+	if err != nil {
+		return fmt.Errorf("unable to delete namespaced multicast objects: %v", err)
+	}
+	return nil
+}
+
 // podAddAllowMulticastPolicy adds the pod's logical switch port to the namespace's
 // multicast port group. Caller must hold the namespace's namespaceInfo object
 // lock.
 func podAddAllowMulticastPolicy(nbClient libovsdbclient.Client, ns string, portInfo *lpInfo) error {
-	return libovsdbops.AddPortsToPortGroup(nbClient, hashedPortGroup(ns), portInfo.uuid)
+	return libovsdbops.AddPortsToPortGroup(nbClient, getMulticastPortGroupName(ns), portInfo.uuid)
 }
 
 // podDeleteAllowMulticastPolicy removes the pod's logical switch port from the
 // namespace's multicast port group. Caller must hold the namespace's
 // namespaceInfo object lock.
 func podDeleteAllowMulticastPolicy(nbClient libovsdbclient.Client, ns string, portUUID string) error {
-	return libovsdbops.DeletePortsFromPortGroup(nbClient, hashedPortGroup(ns), portUUID)
+	return libovsdbops.DeletePortsFromPortGroup(nbClient, getMulticastPortGroupName(ns), portUUID)
+}
+
+// updateStaleMulticastACLsDbIDs updates multicast ACLs that don't have new ExternalIDs set.
+// Must be run before WatchNamespace, since namespaceSync function uses syncNsMulticast, which relies on the new IDs.
+func (oc *DefaultNetworkController) updateStaleMulticastACLsDbIDs() error {
+	// Find stale multicast ACLs
+	legacyAclPred := libovsdbops.GetNoOwnerPredicate[*nbdb.ACL]()
+	legacyACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, legacyAclPred)
+	if err != nil {
+		return fmt.Errorf("unable to find multicast ACLs, cannot update stale data: %v", err)
+	}
+	if len(legacyACLs) == 0 {
+		return nil
+	}
+	mcastMatch := getMulticastACLMatch()
+	updatedACLs := []*nbdb.ACL{}
+	for _, acl := range legacyACLs {
+		var dbIDs *libovsdbops.DbObjectIDs
+		var aclT aclType
+		if acl.Match == mcastMatch {
+			// default deny
+			aclT = policyTypeStrToAclType(acl.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey])
```

Comment:
not really, multicast and default netpol acls have always had this ExternalID, since it is also a part of their "unique ID", because the name is the same for ingress and egress acl